boards/risc-v/litex/arty_a7/src/litex_sdio.c

```diff
@@ -0,0 +1,134 @@
+/****************************************************************************
+ * boards/risc-v/litex/arty_a7/src/litex_sdio.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <debug.h>
+#include <errno.h>
+
+#include <nuttx/sdio.h>
+#include <nuttx/mmcsd.h>
+
+#include "litex.h"
+#include "arty_a7.h"
+#include "litex_sdio.h"
+
+#ifdef HAVE_SDMMC
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static FAR struct sdio_dev_s *g_sdio_dev;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: litex_sdio_card_isr_callback
+ *
+ * Description:
+ *   Check if a card is inserted into the selected MMCSD slot
+ *
+ ****************************************************************************/
+
+static int litex_sdio_card_isr_callback(void *arg)
+{
+  bool cd;
+
+  cd = litex_cardinserted(SDIO_SLOTNO);
+  finfo("Card detect: %d\n", cd);
+  sdio_mediachange(g_sdio_dev, cd);
```

Comment:
why not to get rid of `g_sdio_dev` and pass return of `sdio_initialize` as an `arg` to `litex_sdio_card_isr_callback`?

Reply:
Another good idea. Done.